libgalois/include/katana/GraphTopology.h

```diff
@@ -605,6 +607,153 @@ class KATANA_EXPORT BasicTopologyWrapper : public GraphTopologyTypes {
   const Topo* topo_ptr_;
 };
 
+class KATANA_EXPORT ProjectedShuffleTopology : public GraphTopologyTypes {
```

Comment:
What does this class do? This needs to documentation because it's name implies that it supports some kind of general shuffle operation, but it appears to be type based. Also we already have `BuildOrGetEdgeTypeAwareTopo` and other ones that talk about shuffling. So it's not clear how this is distinct from those.

Reply:
this class creates a new local graph by filtering nodes and edges according to their types. I will add documentation then

Reply:
Please add doxygen comments using `\\\`

Reply:
Also, I think this should just be `ProjectedTopology`. The projection itself can be based on types (for now) or arbitrary Cypher queries (for later).